org.eclipse.xtext/src/org/eclipse/xtext/serializer/impl/Serializer.java

```diff
@@ -210,7 +211,28 @@ public ReplaceRegion serializeReplacement(EObject obj, SaveOptions options) {
 			throw new IllegalStateException("Cannot replace an obj that has no associated node");
 		}
 		String text = serialize(obj, options);
-		return new ReplaceRegion(node.getTotalOffset(), node.getTotalLength(), text);
+		int replaceRegionLength = calculateReplaceRegionLength(node, text);
+		return new ReplaceRegion(node.getTotalOffset(), replaceRegionLength, text);
 	}
 
+	private int calculateReplaceRegionLength(ICompositeNode node, String text) {
+		int oldTextLength = node.getTotalLength();
+		int newTextLength = text.length();
+
+		if (newTextLength > oldTextLength) {
+			String remainingText = text.substring(oldTextLength);
+			if (containsOnlyWhiteSpaces(remainingText) && whiteSpaceFollows(node)) {
+				return newTextLength;
+			}
+		}
+		return oldTextLength;
+	}
+
+	private boolean whiteSpaceFollows(ICompositeNode node) {
+		return node.hasNextSibling() && node.getNextSibling() instanceof HiddenLeafNode;
```

Comment:
Checks against implementation classes of the node model must be avoided. Please use `instanceof ILeafNode && node.isHidden()` instead.

Reply:
In any case, the logic here is unfortunately flawed. If I understand the intention behind this code correctly, it must be something like `node.getLeadNodes().head().isHidden()` - though the name `whiteSpaceFollows` is misleading given that comment nodes are also hidden.

Reply:
Fixed.